airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SqlOperations.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.jdbc;
+
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.util.stream.Stream;
+
+// todo (cgardens) - is it necessary to expose so much configurability in this interface. review if
+// we can narrow the surface area.
+public interface SqlOperations {
+
+  /**
+   * Create a schema with provided name if it does not already exist.
+   *
+   * @param schemaName name of schema.
+   * @throws Exception exception
+   */
+  void createSchemaIfNotExists(String schemaName) throws Exception;
```

Comment:
You could pass the JDBC database object in method with side-effects

Reply:
good point. i'll try this out.

Reply:
i'm not totally happy with this either. because we end up in this scenario where anywhere we pass sql operations we also pass the db. will stick with it for now though.